### PR TITLE
Switch the default behavior of adding labels to OverwriteExistingDstKey

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes https://github.com/openshift/origin/issues/1825

### CODE REVIEW

**Comment by mfojtik** [comment]
@smarterclayton @kargakis PTAL

File: pkg/cmd/cli/cmd/process.go [hunk 1]
```diff
@@ -172,19 +173,14 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
...
-			templateObj.ObjectLabels[key] = value
-		}
-	}
-
 	result, err := client.TemplateConfigs(namespace).Create(templateObj)
 	if err != nil {
 		return err
 	}
 
+	label := kcmdutil.GetFlagString(cmd, "labels")
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/process.go]: I think this needs to happen before we create the resources from the template ie. where the previous labeling was happening.

Comment (kargakis) [inline pkg/cmd/cli/cmd/process.go]: s/create/process

Comment (mfojtik) [inline pkg/cmd/cli/cmd/process.go]: @kargakis why? I think you want to add the labels into the Config (result), not into the template.

Comment (kargakis) [inline pkg/cmd/cli/cmd/process.go]: Well, it probably works for the generated resources in both ways...

Comment (mfojtik) [inline pkg/cmd/cli/cmd/process.go]: @kargakis you can have some label set in the template that will contain parameter. By overriding it, you basically changing the template before it gets chance to process. I think we should be only adding the labels to the result Config/List as the object you are going to create. You are not creating Template, just processing it.

Comment (kargakis) [inline pkg/cmd/cli/cmd/process.go]: I'm convinced, it's safer.

**Comment by kargakis** [comment]
LGTM

**Comment by smarterclayton** [comment]
Please rebase this and add the negative test.

**Comment by smarterclayton** [comment]
Not necessary anymore, handled as part of other template changes

**Pull request closed without merging**